vortex-array/src/array/chunked/flatten.rs

```diff
@@ -0,0 +1,146 @@
+use arrow_buffer::{BooleanBuffer, MutableBuffer, ScalarBuffer};
+
+use vortex_dtype::{DType, match_each_native_ptype, Nullability, PType, StructDType};
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::{Array, ArrayTrait, ArrayValidity, Flattened, IntoArray};
+use crate::accessor::ArrayAccessor;
+use crate::array::bool::BoolArray;
+use crate::array::chunked::ChunkedArray;
+use crate::array::primitive::PrimitiveArray;
+use crate::array::r#struct::StructArray;
+use crate::array::varbin::builder::VarBinBuilder;
+use crate::array::varbin::VarBinArray;
+use crate::validity::{LogicalValidity, Validity};
+
+pub fn try_flatten_chunks(chunks: Vec<Array>, dtype: DType) -> VortexResult<Flattened> {
+    match &dtype {
+        // Structs can have their internal field pointers swizzled to push the chunking down
+        // one level internally without copying or decompressing any data.
+        DType::Struct(struct_dtype, _) => {
+            let struct_array = swizzle_struct_chunks(chunks.as_slice(), struct_dtype)?;
+            Ok(Flattened::Struct(struct_array))
+        }
+
+        // TODO(aduffy): can we pushdown chunks here?
+        DType::Extension(_, _) => {
+            // How can we stitch an unknown type back together?
+            todo!()
+        }
+
+        // Lists just flatten into their inner PType
+        DType::List(_, _) => {
+            todo!()
```

Comment:
Good question... I think one of the next big features is list dtypes. Let's just punt for now